wagtail_newsletter/views.py

```diff
@@ -51,3 +52,40 @@ def dispatch(self, request: HttpRequest, *, page_id: int, revision_id: int):  #
 
     def get(self, request):
         return self.render_to_response(self.get_context_data())
+
+    def post(self, request):
+        try:
+            campaign_id = self.backend.save_campaign(
+                campaign_id=self.object.newsletter_campaign,
+                **self.campaign_data,
+            )
+
+        except campaign_backends.CampaignBackendError:
+            messages.error(request, "Failed to save campaign")
+            return redirect(".")
+
+        self.object.newsletter_campaign = campaign_id
+        self.object.save(update_fields=["newsletter_campaign"])
+
+        try:
+            campaign = self.backend.get_campaign(campaign_id)
+            if campaign is None:  # pragma: no cover
+                raise RuntimeError("The campaign we just created is not there any more")
```

Comment:
This error should never happen, so I haven't handled it nicely, nor written tests for it.